ranger/default.go

```diff
@@ -251,7 +270,13 @@ func DefaultResponder(opts ...resp.ResponderOptFn) RangerOption {
 			}
 
 			if rng.p == nil {
-				if _, err := DefaultParser(os.DirFS("."))(rng); err != nil {
+				vfs := &virtualFS{
+					cache:  make(map[string]func(string) (fs.File, error)),
+					osDir:  os.DirFS("."),
+					pkgDir: pkgFS,
+					Mutex:  sync.Mutex{},
+				}
+				if _, err := DefaultParser(vfs)(rng); err != nil {
```

Comment:
Curious if this implementation of the vfs should live in http/template rather than being tried to ranger

Reply:
That's a great question, I hadn't considered it due to the `go:embed` directive in `ranger/fs`. Again, the constraint with `embed` is wherever the embedded files live, the embed directive must be in the same directory as those files or a direct parent of them.  We see this with the embed directive existing in `ranger/` and the files in `ranger/tmpl/`. So, moving it to `http/template` would also mean moving the files there, for example `http/template/default`

This isn't a problem - just a question of where it makes more sense to have those templates live. Definitely not opposed to it, either. If we always use the `ranger` package to include defaults, I can see it getting messy. I'm thinking, for example, of the desire to move Sentry logging out of `logger`. Curious if these comments clarify the idea for you.

Reply:
Okay, I played around with it being in `http/template` and concluded against moving it there in favor of keeping it in `ranger` as a default. The main reason was the functions and data the template needs are outside the scope of `http/template`.

Here's where my thinking is heading. `ranger` is the single dumping ground for defaults values, configuration, etc. rather than the packages it imports. This ways, there's a single layer someone can review when behavior doesn't match expectations instead of defaults living in different layers. Don't like the default error template? Look at `ranger`. Don't like the default environment? Look at `ranger`. And so forth. `ranger` functions as the "glue" or "spaghetti" code between the other `trails` packages so their respective scopes stay focused.